tutorials/smart-contracts/launch-your-first-project/create-contracts.md

```diff
@@ -1,6 +1,150 @@
+- **Functions** - executable code that can read or modify the contract's state variables
+- **Events** - notification mechanisms that applications can subscribe to in order to track blockchain changes
+
+## Create the Smart Contract
+
+In this section, you'll build a simple storage contract step by step. Later, you'll explore each component in more detail to understand what's happening behind the scenes.
+
+This contract will:
+
+- Store a number
+- Allow updating the stored number
+- Emit an event when the number changes
+
+To build the smart contract, follow the steps below:
+
+1. Create a new file named `Storage.sol` in your text editor
+
+2. Add the SPDX license identifier at the top of the file. The Solidity compiler requires this:
+
+    ```solidity
+    // SPDX-License-Identifier: MIT
+    ```
+
+    This line tells users and tools which license governs your code. The [MIT license](https://opensource.org/license/mit){target=\_blank} is commonly used for open-source projects. The Solidity compiler requires this line to avoid licensing-related warnings.
+
+3. Specify the Solidity version you want to use. We'll use a recent stable version:
+
+    ```solidity
+    pragma solidity ^0.8.19;
+    ```
+
+    The caret `^` means "this version or any compatible newer version." This helps ensure your contract compiles correctly with the intended compiler features.
+
+4. Create the contract structure:
+
+    ```solidity
+    contract Storage {
+        // Contract code will go here
+    }
+    ```
+
+    This defines a contract named "Storage", similar to how you would define a class in other programming languages.
+
+5. Add the state variables and event:
+
+    ```solidity
+    contract Storage {
+        // State variable to store a number
+        uint256 private number;
+        
+        // Event to notify when the number changes
+        event NumberChanged(uint256 newNumber);
+    }
+    ```
+
+    Here, you're defining:
+
+    - A state variable named `number` of type `uint256` (unsigned integer with 256 bits), which is marked as `private` so it can only be accessed via functions within this contract
+    - An event named `NumberChanged` that will be triggered whenever the number changes. The event includes the new value as data
+
+6. Add the getter and setter functions:
+
+    ```solidity
+    --8<-- 'code/tutorials/smart-contracts/launch-your-first-project/create-contracts/Storage.sol'
+    ```
+
+The complete contract should look like this:
+
+??? code "Storage.sol"
+
+    ```solidity title="Storage.sol"
+    --8<-- 'code/tutorials/smart-contracts/launch-your-first-project/create-contracts/Storage.sol'
+    ```
+
+## Understanding the Code
+
+Let's break down the key components of the contract:
+
+- **State Variable**
+
+    - `uint256 private number` - a private variable that can only be accessed through the contract's functions
+    - The `private` keyword prevents direct access from outside the contract
```

Comment:
Maybe explain a little bit more that this variable cannot be accessed directly from other contracts outside this one, but the data is still public and can be read by external tools